doc/man3/flux_future_promise.adoc

```diff
@@ -0,0 +1,163 @@
+flux_future_promise(3)
+======================
+:doctype: manpage
+
+
+NAME
+----
+flux_future_promise, flux_future_set_result, flux_future_set_error, flux_future_aux_get, flux_future_aux_set, flux_future_set_flux, flux_future_get_flux - support methods for future wrappers
+
+
+SYNOPSIS
+--------
+ #include <flux/core.h>
+
+ typedef void (*flux_promise_f)(flux_future_t *f,
+                                flux_reactor_t *r, void *arg);
+
+ void flux_future_promise (flux_future_t *f,
+                           flux_promise_f cb, void *arg);
+
+ void flux_future_set_result (flux_future_t *f,
+                              void *result, flux_free_f free_fn);
+
+ void flux_future_set_error (flux_future_t *f, int errnum);
+
+ void *flux_future_aux_get (flux_future_t *f, const char *name);
+
+ int flux_future_aux_set (flux_future_t *f, const char *name,
+                          void *aux, flux_free_f destroy);
+
+ void flux_future_set_flux (flux_future_t *f, flux_t *h);
+
+ flux_t *flux_future_get_flux (flux_future_t *f);
+
+
+DESCRIPTION
+-----------
+
+A Flux future represents some activity that may be completed with reactor
+watchers and/or message handlers.  It is intended to be wrapped
+by other classes that perform specific activities, providing a common idiom
+for synchronizing on the result.  Typically such a class provides
+class-specific wrappers for `flux_future_create()` and `flux_future_get()`,
+and implements a class-specific promise callback.
+
+`flux_future_promise()` registers a promise callback.  The promise sets
+up class-specific watchers on the reactor to fulfill the promise as
+asynchronous events occur.  The watchers must eventually call
```

Comment:
No I think you are right here - I should use consistent terminology.  I'll go with "fulfill the future".  (I didn't really mean to introduce a separate concept "fullfill the promise" - I was using that interchangeably with "complete the future").  Good catch.